packages/core/src/repository/data-source/data-source-mapper.ts

```diff
@@ -12,35 +12,35 @@ import { DeviceConsoleLogger, Logger } from '../../helpers';
  * @param toInMapper Mapper to map repository objects to data source objects
  */
 export class DataSourceMapper<In, Out> implements GetDataSource<Out>, PutDataSource<Out>, DeleteDataSource {
+    private readonly getMapper: GetDataSourceMapper<In, Out>;
+    private readonly putMapper: PutDataSourceMapper<In, Out>;
+
     constructor(
-        private readonly getDataSource: GetDataSource<In>,
-        private readonly putDataSource: PutDataSource<In>,
+        getDataSource: GetDataSource<In>,
+        putDataSource: PutDataSource<In>,
         private readonly deleteDataSource: DeleteDataSource,
-        private readonly toOutMapper: Mapper<In, Out>,
-        private readonly toInMapper: Mapper<Out, In>,
+        toOutMapper: Mapper<In, Out>,
+        toInMapper: Mapper<Out, In>,
```

Comment:
Why removing the access modifier and the readonly?

Reply:
I think the code is correct without `private readonly`. `private readonly` only makes sense if these parameters are going to be used anywhere else in the class. Inside the constructor `readonly` doesn't give any guarantee ([see docs](https://www.typescriptlang.org/docs/handbook/2/classes.html#readonly)):

> Fields may be prefixed with the `readonly` modifier. This prevents assignments to the field **outside of the constructor**.

So, inside the constructor these can be reasigned.

Reply:
In the end is like with any other function. I agree with deleting this is the scope of this parameters is only the constructor.